### PR TITLE
pythonPackages.scikit-build: init at 0.8.1

### PR DESCRIPTION
**Pull request by FlorianFranzen**
###### Motivation

Tool to build easily build Python extensions with CMake.

###### Things done

Included patch was submitted upstream and is part of next release.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
> Could not find suitable distribution for Requirement.parse('pytest-runner<3dev,>=2.0')

You will need to patch the source to allow a more recent version of pytest-runner.

File: pkgs/development/python-modules/scikit-build/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/scikit-build/default.nix]: No period: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/development/python-modules/scikit-build/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/scikit-build/default.nix]:
```suggestion
    license = with lib.licenses; [ mit bsd2 ];
```
See https://github.com/scikit-build/scikit-build/blob/master/LICENSE. Also add a comment explaining why there are two.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by FlorianFranzen** [comment]
I updated the expression, but the tests now triggers a bug in the way python3 is build at the moment.

On an older NixOS system:
```
$ python3 -c 'import sysconfig; print(sysconfig.get_config_var("INCLUDEPY"))' 
/nix/store/hy65mn4wjswqih75gfr6g4q3xgqdm325-python3-3.6.6/include/python3.6m
```

On the current unstable channel:
```
$ nix-shell --pure -p python36 --run "python3 -c 'import sysconfig; print(sysconfig.get_config_var(\"INCLUDEPY\"))'"
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-python3-3.6.8/include/python3.6m
```
I will triage the source of the issue a bit longer, but it seems like ```nuke-references``` breaks ```sysconfig``` at the moment.

**Comment by dotlambda** [comment]
That's fixed by https://github.com/NixOS/nixpkgs/pull/54486.

**Comment by FlorianFranzen** [comment]
So I guess we will have to wait for staging to be merged into master, before we can run these test again. Not to mention that I am still waiting to be added as a maintainer in a previous pull-request.

**Comment by dotlambda** [comment]
You can also make your PR against staging.

**Comment by FlorianFranzen** [comment]
@dotlambda: True, that is probably the easiest.

Now with ```staging-next``` as base and the missing maintainer entry.

File: pkgs/development/python-modules/scikit-build/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/scikit-build/default.nix]:
```suggestion
    license = [ licenses.mit licenses.bsd2 ]; # BSD due to reuses of PyNE code
```
wor use `with licenses;`

Comment (FlorianFranzen) [reply pkgs/development/python-modules/scikit-build/default.nix]: Not misspelling it would be a good start. Sorry for that. Fixed and forced.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.scikit-build python3.pkgs.scikit-build

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.scikit-build

File: pkgs/development/python-modules/scikit-build/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/scikit-build/default.nix]: It's still incorrect.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Btw, adding yourself to `maintainer-list.nix` should be a separate commit.

**Comment by FlorianFranzen** [comment]
I split the commit in two and added the missing spelling correction. 

Seems however like I have upset @GrahamcOfBorg.

**Comment by dotlambda** [comment]
@FlorianFranzen Thank you for your patience!

@GrahamcOfBorg build python2.pkgs.scikit-build

**Comment by FlorianFranzen** [comment]
@dotlambda No problem. I was the one that pushed a broken commit.

The Python 3 builds are the most important ones, as the scikit-build testsuite will tell if Python's sysconfig is working (fixed by #54486, which was merged into unstable-next).

**Pull request merged**
